hpx/util/pack_traversal.hpp

```diff
@@ -0,0 +1,729 @@
+//  Copyright (c) 2017 Denis Blank
+//
+//  Distributed under the Boost Software License, Version 1.0. (See accompanying
+//  file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+#ifndef HPX_UTIL_PACK_TRAVERSAL_HPP
+#define HPX_UTIL_PACK_TRAVERSAL_HPP
+
+#include <type_traits>
+#include <utility>
+
+#include <hpx/config.hpp>
+#include <hpx/traits/is_callable.hpp>
+#include <hpx/traits/is_range.hpp>
+#include <hpx/traits/is_tuple_like.hpp>
+#include <hpx/util/always_void.hpp>
+#include <hpx/util/invoke.hpp>
+#include <hpx/util/invoke_fused.hpp>
+#include <hpx/util/result_of.hpp>
+#include <hpx/util/tuple.hpp>
+
+/// \cond false
+// TODO Find a better solution for this
+#if !defined(HPX_MSVC)
+#define HPX_HAVE_EXPRESSION_SFINAE
+#endif
+/// \endcond
+
+namespace hpx {
+namespace util {
+    namespace detail {
```

Comment:
We should probably avoid having doxygen to generate docs for this, the detail namespace is not for public consumption anyway.